### PR TITLE
Refactor actor parsing

### PR DESCRIPTION
**Pull request by Longi94**
Separate classes to handle different types of actors. It makes the code cleaner and easier to maintain. Also comes with a slight speed up. Parsing a full replay took ~0.1 seconds less time (1.778 -> 1.677) and full pytest took 3 seconds less time (172.50 -> 169.46) on my local machine. This is a big PR, I've yet to test if the parser generates an identical dataframe as the old implementation, shouldn't be merged before I (or someone else) do that.

- [x] Add test for ensuring that dataframes stay identical across changes

### CODE REVIEW

**Comment by dtracers** [comment]
After this it goes to another set of python code to create the protobuf.
Do you think that this is in it would be possible to create the protobuf and dataframe all during python parsing?

It could save some lines of code but idk if it will increase performance noticeably

**Comment by Longi94** [comment]
In theory, yes it should be possible for staistics/events that can be calculated incrementally. We could avoid having to create and store that large dataframe and making slow calculations on that. I can't think of one now, but maybe there are some metrics that would need data from every frame.

**Comment by dtracers** [comment]
We need the dataframe since we do a ton of metrics on it. Things like boosts, where you are on the field all depend on the dataframe. but if we did parsing in faster language then we could avoid that issue.

But that might not get started on that until we are in a much better place.
This will be really useful though for making those changes easier.

File: carball/json_parser/actor/game_event.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from .base import *
+
+
+class GameEventHandler(BaseActorHandler):
+    type_name = 'Archetypes.GameEvent.GameEvent_Soccar'
```
Comment (twobackfromtheend) [inline carball/json_parser/actor/game_event.py]: Is this used for anything? Or just here for future reference.

File: carball/json_parser/actor/base.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+COMPONENT_ACTIVE_KEY = 'TAGame.CarComponent_TA:Active'
+
+
+class BaseActorHandler(object):
+    """
+    Base class for actor handlers.
+
+    If can_handle is not overridden, it will check the type_name attribute. The priority attribute determines the order
+    of the handlers. Lower number means higher priority. If the priority is the same, the order will depend on the order
+    of actors int he replay.
```
Comment (twobackfromtheend) [inline carball/json_parser/actor/base.py]: in the* replay

also I'd put the comments for the two attributes on different lines.

File: carball/json_parser/actor/game_event.py (hunk 1 above)
Comment (Longi94) [reply carball/json_parser/actor/game_event.py]: I probably left it there by accident

**Comment by twobackfromtheend** [comment]
Is there any work left on this or is it pretty much done?

**Comment by Longi94** [comment]
Ah sorry, I got distracted, I still need to add those small changes in you comment. I'm assuming there's now a test to ensure identical dataframes (because of the checkbox? Otherwise the only thing I'd do is merge master into this branch to make sure that doesn't break things.

**Comment by Longi94** [comment]
Should be good now. Is the checkbox you added to the description done?

**Comment by dtracers** [comment]
I have not done anything to ensure that the frames are identical.
@twobackfromtheend did you preform this check?

**Comment by dtracers** [comment]
I'm not too concerned since it will be changing with rust potentially as well

**Pull request merged**